### PR TITLE
Quantum Go

### PR DESCRIPTION
**Pull request by benjaminpjones**
UPDATE (3/24): Implementation should be complete!  Feel free to leave feedback :)

https://github.com/govariantsteam/govariants/assets/25233703/784be9f5-8e29-4b8d-b871-20de82412f62

---

As of Feb 19, it's not functional at all, but this is the boilerplate that needs to be in place to get a variant added.  It just shows two boards with a quantum stone hard-coded at the 3-4 point:

Screenshot 2024-02-19 at 1 09 54 AM

### CODE REVIEW

**Comment by benjaminpjones** [comment]
Apologies for the force push, I think I forgot to include a new file in one of the commits and then rebased.

Implementation should be complete, so moving to "Ready for review"!

File: packages/shared/src/variants/quantum.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    return {
+      boards: this.subgames.map((game) => game.badukGame.board.to2DArray()),
+      quantum_stones: this.subgames[0].quantumMappings.map((mapping) =>
+        mapping.map((pos) => pos.toSgfRepr()),
+      ),
+    };
+  }
+  nextToPlay(): number[] {
+    return [this.round % 2];
```
Comment (merowin) [inline packages/shared/src/variants/quantum.ts:186]: Please return empty array when the game is finished.

**Review by merowin — COMMENTED** [review]
Looks great! Can't wait to try it out :-)

File: packages/shared/src/variants/quantum.ts (hunk 1 above)
Comment (benjaminpjones) [reply packages/shared/src/variants/quantum.ts:186]: Will do!

And whoops... I am realizing I forgot to implement endgame.  Sending back to draft

Comment (benjaminpjones) [reply packages/shared/src/variants/quantum.ts:218]: Fixed!

**Review by merowin — APPROVED** [review]
nice!

**Pull request merged**